docs/src/main/asciidoc/extension-maturity-model.adoc

```diff
@@ -0,0 +1,87 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+[id="extension-maturity-model"]
+= A maturity model for Quarkus extensions
+include::_attributes.adoc[]
+:diataxis-type: concept
+:categories: writing-extensions
+:topics: extensions
+:summary: Quarkus extensions can do a lot, or a little. This guide explains some of the capabilities extension authors might want to include.
+////
+The document header ends at the first blank line. Do not remove the blank line between the header and the abstract summary.
+////
+
+What makes a good Quarkus extension? What capabilities is a Quarkus extension expected to provide?
+
+image::extension-maturity-model.svg[A maturity model: Runtime actually works,Works in JVM mode,Works in dev mode, Works in native,Configuration support,Dev service,Dev UI, Use build-time application knowledge to eliminate wasteful runtime code paths (supersonic subatomic performance),Use build-time application knowledge to remove boilerplate,Codestart application template]
```

Comment:
what does the bracketed smiley face indicate?

Reply:
Developer joy. So the idea is that there are several different things extension owners should do in the theme of "developer joy". Now that we have one, we could also take the "performance" icon from https://github.com/quarkusio/quarkusio.github.io/pull/2122 and put that on the performance-related items.

Reply:
It took me way more time than I care to admit that that icon is the same as the one shown on https://quarkus.io/developer-joy/ :)

Could we somehow link/highlight that - to somehow guide daft readers (like me :) to make the connection.

Reply:
I'm hoping it will be a lot more clear once we have a performance icon we can annotate the performance items with, so then people will recognise two items from the front page - but that needs https://github.com/quarkusio/quarkusio.github.io/pull/2122 to define a performance icon, and then it needs the front page updates, and all that work is still in progress. So I don't think it makes sense to block this content on that work.

Reply:
I'd suggested focussing on the 'what', not the 'how', to tie it back to the front page. (Which also should focus on the benefits to users, not the 'how'.) So categories should be something like

 - developer joy - dev UI, hot reload, dev services, unified config, boilerplate reduction, programming model simplification
-  performance - goes fast, uses very few resources (and the ingredients of this is the reactive core and the build first principle, but those are how, not what)
 -  prod/cloud readiness - observability, manageability, operations concerns